### PR TITLE
Add distributor factory

### PR DESCRIPTION
**Pull request by prevostc**
**JIRA task url** (if applicable)

**(If Bug) What was the problem being solved?**

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

**How to reproduce fix / test this PR?**

### CODE REVIEW

File: v2/airdrop-platform/contracts/hardhat.config.ts [hunk 1]
```diff
@@ -25,18 +25,28 @@ const config = {
...
+          browserURL: 'https://amoy.polygonscan.com',
+        },
+      },
+    ],
   },
   sourcify: {
     // Disabled by default
     // Doesn't need an API key
-    enabled: true,
+    enabled: false,
```
Comment (joelamouche) [inline v2/airdrop-platform/contracts/hardhat.config.ts:49]: why this change?

**Review by joelamouche — APPROVED** [review]
LGTM!
One minor question and also what I saud about making the distributor ownable and rescue-able

File: v2/airdrop-platform/contracts/hardhat.config.ts (hunk 1 above)
Comment (prevostc) [reply v2/airdrop-platform/contracts/hardhat.config.ts:49]: verification on sourcify.dev didn't work, had to disable it
